src/Test.UnitTests.Sarif/Core/AddressTests.cs

```diff
@@ -0,0 +1,227 @@
+    {
+        private const string AbsoluteAddress = "\"AbsoluteAddress\"";
+        private static readonly TestAssetResourceExtractor s_extractor = new TestAssetResourceExtractor(typeof(AddressTests));
+
+        [Fact]
+        public void Address_VerifyRoundTripFromObject()
+        {
+            // Arrange
+            var testCases = new[] {
+                new { absoluteAddress = (BigInteger?)null, expectedShouldSerializeAbsoluteAddress = false },
+                new { absoluteAddress = (BigInteger?)long.MinValue - 1, expectedShouldSerializeAbsoluteAddress = false },
+                new { absoluteAddress = (BigInteger?)long.MinValue, expectedShouldSerializeAbsoluteAddress = false },
+                new { absoluteAddress = (BigInteger?)int.MinValue - 1, expectedShouldSerializeAbsoluteAddress = false },
+                new { absoluteAddress = (BigInteger?)int.MinValue, expectedShouldSerializeAbsoluteAddress = false },
+                new { absoluteAddress = (BigInteger?)(-2), expectedShouldSerializeAbsoluteAddress = false },
+                new { absoluteAddress = (BigInteger?)(-1), expectedShouldSerializeAbsoluteAddress = false },
+                new { absoluteAddress = (BigInteger?)0, expectedShouldSerializeAbsoluteAddress = true },
+                new { absoluteAddress = (BigInteger?)1, expectedShouldSerializeAbsoluteAddress = true },
+                new { absoluteAddress = (BigInteger?)2, expectedShouldSerializeAbsoluteAddress = true },
+                new { absoluteAddress = (BigInteger?)int.MaxValue, expectedShouldSerializeAbsoluteAddress = true },
+                new { absoluteAddress = (BigInteger?)int.MaxValue + 1, expectedShouldSerializeAbsoluteAddress = true },
+                new { absoluteAddress = (BigInteger?)long.MaxValue, expectedShouldSerializeAbsoluteAddress = true },
+                new { absoluteAddress = (BigInteger?)long.MaxValue + 1, expectedShouldSerializeAbsoluteAddress = true },
+                new { absoluteAddress = (BigInteger?)ulong.MaxValue, expectedShouldSerializeAbsoluteAddress = true },
+                new { absoluteAddress = (BigInteger?)ulong.MaxValue + 1, expectedShouldSerializeAbsoluteAddress = true },
+            };
+
+            var address = new Address();
+            foreach (var testCase in testCases)
+            {
+                if (testCase.absoluteAddress.HasValue)
+                {
+                    address.AbsoluteAddress = testCase.absoluteAddress.Value;
+                }
+
+                BigInteger expectedAbsoluteAddress =
+                    (!testCase.absoluteAddress.HasValue || testCase.absoluteAddress.Value < 0) ? -1 : testCase.absoluteAddress.Value;
+
+                // Act & Assert
+                Address_VerifyRoundTripFromObjectHelper(address,
+                    expectedShouldSerializeAbsoluteAddress: testCase.expectedShouldSerializeAbsoluteAddress,
+                    expectedAbsoluteAddress: expectedAbsoluteAddress);
```

Comment:
>expectedAbsoluteAddress: expectedAbsoluteAddress [](http://example.com/codeflow?start=20&length=48)

you don't need to repeat yourself

Reply:
this applies for the other tests as well.

Reply:
Thanks for the explanation in the other comment.